src/main/grammar/SourcePawn.flex

```diff
@@ -5,26 +5,37 @@ import static org.idea_sp.psi.SourcePawnTypes.*;
 
 %%
 
-%{
-  public _SourcePawnLexer() {
-    this((java.io.Reader)null);
-  }
-%}
-
 %public
 %class _SourcePawnLexer
 %implements FlexLexer
 %function advance
 %type IElementType
 %unicode
+%line
+%column
+
+%{
+  public _SourcePawnLexer() {
+    this((java.io.Reader)null);
+  }
+
+  int yyline;
+  int yycolumn;
+  int commentStart;
+  IElementType commentType;
+%}
 
 EOL="\r"|"\n"|"\r\n"
 LINE_WS=[\ \t\f]
 WHITE_SPACE=({LINE_WS}|{EOL})+
 
 SPACE=[\ \n\r\t\f]
 LINE_COMMENT="//"[^\r\n]*
-BLOCK_COMMENT="/"\*(.|\n)*?\*"/"
+BLOCK_COMMENT=[\s\S]
+DOC_COMMENT=[\s\S\n]
```

Comment:
Isn't `[^]` the same as `[\s\S\n]`? Also `\s` should include'\n`

Reply:
I don't believe I had a good grasp of the purpose of these definitions when I wrote them. I think I believed they became token or element types, so I thought it was necessary to define them in order to `return` them for use in the parser... but that appears untrue. Instead, the grammar's `tokens` definition seems to be the source of truth.

The regexes defined for `BLOCK_COMMENT` and `DOC_COMMENT` are never used, since we just look for their starting sequences and consume all tokens till their ending sequences. (Though, I guess the token consumption _does_ use `[\s\S\n]`.) Those definitions should be removed for clarity.

I had assumed `[\s\S]` would include all characters; however, I found that after adding `\n` to the character group, it properly parsed multiline comments. I wasn't really scientific enough about it to say whether one caused the other, and not something else I'd done... Regardless `[^]` is definitely much more succinct. I wish _I'd_ thought of that :P (Or looked down to the `BAD_CHARACTER` rule)